### DISCUSSION TITLE
MultiAuthEvents-class is missing the IAuthEventsAsync implementation

### DISCUSSION THREAD

**Discussion by ItsClemi** [ServiceStack/Discuss#52] [Ideas]
Hi,

we recently switched our auth-features to use the `IAuthEventsAsync`-interface. As we're using multiple auth-event features, they are internally combined by the `MultiAuthEvents` class. This class only implements the `IAuthEvents` interface but is missing the `IAuthEventsAsync` interface, so event though all `*Async` methods are implemented none are called as the interface cast is always false.

For now we're using a workaround with a custom `MultiAuthEvents` class which simply implements both `MultiAuthEvents, IAuthEventsAsync` and a custom `AuthFeature` feature which uses our custom `MultiAuthEvents`-class instead of the internal one.

Thanks!

**Comment by mythz** [comment]
Yeah makes sense to implement it on `MultiAuthEvents` also which is done in [this commit](https://github.com/ServiceStack/ServiceStack/commit/8edeef030c19d6c4350123a075794267c655a21e).

Unfortunately we've had several issues over the last few days trying to publish our pre-release packages to MyGet which has been unavailable all day today as right now none of our published packages are no showing up on our MyGet feed. 

To work around this we've also published our pre-release packages to GitHub Packages:
https://github.com/orgs/ServiceStack/packages?repo_name=ServiceStack

Unfortunately GitHub adds more friction which requires a [Personal Access Token with read:packages](https://docs.github.com/en/packages/learn-github-packages/introduction-to-github-packages#authenticating-to-github-packages) so your **NuGet.config** would need to look like:

```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
    <packageSources>
        <clear />
        <add key="github" value="https://nuget.pkg.github.com/ServiceStack/index.json" />
    </packageSources>
    <packageSourceCredentials>
        <github>
            <add key="Username" value="USERNAME" />
            <add key="ClearTextPassword" value="TOKEN" />
        </github>
    </packageSourceCredentials>
</configuration>
```

Where TOKEN is a PAT you've created with at least `read:packages` scope.
